src/apps/service/apps.clj

```diff
@@ -219,17 +219,18 @@
 
 (defn update-job-status
   ([external-id]
+   (jobs/validate-job-status-update-step-count external-id)
    (transaction
-    (let [{job-id :job_id} (jobs/get-unique-job-step external-id)
-          updates          (jp/get-job-status-updates external-id)]
+    (let [updates (jp/get-job-status-updates external-id)]
       (when (seq updates)
-        (let [job-step    (jobs/lock-job-step job-id external-id)
+        (let [job-id      (:job_id (jp/get-job-steps-by-external-id external-id))
```

Comment:
It seems like `get-job-steps-by-external-id` will return a list of results?
Should `first` be called on the results before `:job_id`?

Reply:
Ah, yes it should. Thanks for catching that.